### PR TITLE
Add --openssl-repo option for downloading OpenSSL source

### PR DESCRIPTION
**Pull request by pshipton**
I'll port this to other versions after review.

### CODE REVIEW

File: get_source.sh [hunk 1]
```diff
@@ -47,6 +47,7 @@ usage() {
 	echo "  -omr-sha          a commit SHA for the omr repository"
 	echo "  -omr-reference    a local repo to use as a clone reference"
 	echo "  -parallel         (boolean) if 'true' then the clone j9 repository commands run in parallel, default is false"
+	echo "  --openssl-repo    Specify the openssl repo to download from"
```
Comment (keithc-ca) [inline get_source.sh:50]: In the help text, please spell OpenSSL consistently and repository in full.

File: closed/get_openssl_source.sh [hunk 2]
```diff
@@ -41,6 +42,10 @@ do
 		usage
 		;;
 
+		--openssl-repo=* )
+		GIT_URL="${i#*=}"
```
Comment (keithc-ca) [inline closed/get_openssl_source.sh:46]: Suggest `OPENSSL_URL` is a better name than `GIT_URL`.

File: closed/get_openssl_source.sh [hunk 3]
```diff
@@ -26,6 +26,7 @@ usage() {
 	echo "Usage: $0 [-h|--help] [--openssl-version=<openssl version 1.0.2 and above to download>]"
```
Comment (keithc-ca) [inline closed/get_openssl_source.sh:26]: Perhaps this should change; either mention the new option or just say `[options]` or similar.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

**Comment by pshipton** [comment]
Updated to address review comments.

**Comment by pshipton** [comment]
Pushed another update to change the order of the Usage options.

**Comment by pshipton** [comment]
Oops, pushed another update to fully replace GIT_URL

File: closed/get_openssl_source.sh [hunk 4]
```diff
@@ -23,16 +23,17 @@
 # ===========================================================================
 
 usage() {
-	echo "Usage: $0 [-h|--help] [--openssl-version=<openssl version 1.0.2 and above to download>]"
+	echo "Usage: $0 [-h|--help] [--openssl-repo=<repo URL>] [--openssl-version=<openssl version 1.0.2 and above to download>]"
 	echo "where:"
 	echo "  -h|--help             print this help, then exit"
+	echo "  --openssl-repo        OpenSSL repo. By default, https://github.com/openssl/openssl.git"
```
Comment (keithc-ca) [inline closed/get_openssl_source.sh:29]: This should use the full word "repository" as well.

Comment (pshipton) [reply closed/get_openssl_source.sh]: Updated.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
jenkins compile xlinux jdk8

**Pull request merged**